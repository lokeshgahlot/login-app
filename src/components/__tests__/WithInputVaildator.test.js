import React from "react";
import {shallow} from "enzyme"
import WithInputVaildator from "../WithInputValidator";

describe("<WithInputVaildator />", () => {
  const message = "Invalid email";
  // Todo: Couldn't better way to test this scenario
  //Skipping for now 
  it.skip ("should throw Error when zero or more than one children", () => {
    // const wrapper = shallow(
    //   <WithInputVaildator>
    //     <input />
    //     <span>hello</span>
    //   </WithInputVaildator>
    // );
    // expect(wrapper).toThrow(new Error("<WithEmailVaildator /> should have one child only!"));
  });

  it ("renders correctly", () => {
    const wrapper = shallow(
        <WithInputVaildator>
          <input />
        </WithInputVaildator>
      );
  });

  it ("calls onChange", () => {
    const fn = jest.fn();
    const wrapper = shallow(
        <WithInputVaildator>
          <input className="custom-input" onChange={fn}/>
        </WithInputVaildator>
      );
    wrapper.find(".custom-input").simulate("change", {
      target: {
        value: "test"
      }
    });
    expect(fn.mock.calls.length).toBe(1);
  });

  it ("onChange() calls setCustomValidity", () => {
    const fn = jest.fn();
    const mockSetCustomValidity = jest.fn();
    const wrapper = shallow(
        <WithInputVaildator>
          <input className="custom-input" onChange={fn}/>
        </WithInputVaildator>
      );
    wrapper.find(".custom-input").simulate("change", {
      target: {
        value: "test@test.com",
        setCustomValidity: mockSetCustomValidity
      }
    });
    expect(mockSetCustomValidity.mock.calls[0][0]).toBe("");

    mockSetCustomValidity.mockClear();
    wrapper.find(".custom-input").simulate("change", {
      target: {
        value: "test@test",
        setCustomValidity: mockSetCustomValidity
      }
    });
    expect(mockSetCustomValidity.mock.calls[0][0]).toBe(message);
  });

  it ("calls onInvalid", () => {
    const fn = jest.fn();
    const mockSetCustomValidity = jest.fn();
    const wrapper = shallow(
        <WithInputVaildator>
          <input className="custom-input" onInvalid={fn}/>
        </WithInputVaildator>
      );
    wrapper.find(".custom-input").simulate("invalid", {target: {}});
    expect(fn.mock.calls.length).toBe(1);
  });

  it ("onInvalid() calls setCustomValidity", () => {
    const fn = jest.fn();
    const mockSetCustomValidity = jest.fn();
    const wrapper = shallow(
        <WithInputVaildator>
          <input className="custom-input" onInvalid={fn}/>
        </WithInputVaildator>
      );
    wrapper.find(".custom-input").simulate("invalid", {
      target: {
        setCustomValidity: mockSetCustomValidity
      }
    });
    expect(mockSetCustomValidity.mock.calls[0][0]).toBe(message);
  });
});