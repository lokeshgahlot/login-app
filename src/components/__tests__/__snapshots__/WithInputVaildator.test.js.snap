// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WithInputVaildator /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithInputVaildator
    message="Invalid email"
    pattern="^([\\\\w.%+-]+)@([\\\\w-]+\\\\.)+([\\\\w]{2,})$"
    type="email"
  >
    <input />
  </WithInputVaildator>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          onInvalid={[Function]}
          pattern="^([\\\\w.%+-]+)@([\\\\w-]+\\\\.)+([\\\\w]{2,})$"
          type="email"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": ".0",
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "onInvalid": [Function],
          "pattern": "^([\\\\w.%+-]+)@([\\\\w-]+\\\\.)+([\\\\w]{2,})$",
          "type": "email",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            onInvalid={[Function]}
            pattern="^([\\\\w.%+-]+)@([\\\\w-]+\\\\.)+([\\\\w]{2,})$"
            type="email"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": ".0",
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "onInvalid": [Function],
            "pattern": "^([\\\\w.%+-]+)@([\\\\w-]+\\\\.)+([\\\\w]{2,})$",
            "type": "email",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
